src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2983,7 +3033,9 @@ QueryTreeNodePtr QueryAnalyzer::tryResolveIdentifierFromJoin(const IdentifierLoo
 
     JoinKind join_kind = from_join_node.getKind();
 
-    if (left_resolved_identifier && right_resolved_identifier)
+    if (left_resolved_identifier && right_resolved_identifier
```

Comment:
Then we use identifier from left table. Since function cannot be under `USING` (https://github.com/ClickHouse/ClickHouse/blob/07de815d96afd94fc659b3b7f4566b5ecef87256/src/Analyzer/Passes/QueryAnalysisPass.cpp#L6531) it should be ok. I added check`single_join_prefer_left_table` for this code branch, I hope it's enough (I haven't found cases where it doesn't work).